testing/sdk_hosts.py

```diff
@@ -43,6 +45,17 @@ def autoip_host(service_name, task_name, port=-1):
         port)
 
 
+def custom_host(service_name, task_name, custom_domain, port=-1):
+    """
+    Returns a properly constructed hostname for the container of the given task using the
+    supplied custom domain.
+    """
+    return _to_host(
```

Comment:
Thinking of it again, would it not make sense to overload `autoip_host` with a `custom_domain` instead of introducing an entirely new function:
```
def autoip_host(service_name, task_name, port=-1, domain=AUTOIP_HOST_SUFFIX):
    return _to_host(_safe_name(task_name), _safe_name(service_name), domain, port)
```

Reply:
As a side note, the `_to_host` could be changed to something like:
```python
def _to_host(*args, port=-1):
     host = ".".join(str(a).strip(".") for a in args)
     if port > -1:
         host = ":".join([host, str(port)])
     return host
```
to allow arbitrary parts.